src/s2_geo.rs

```diff
@@ -0,0 +1,120 @@
+use cgmath::Vector2;
+use nav_types::{ECEF, WGS84};
+use s2::cap::Cap;
+use s2::cellid::CellID;
+use s2::latlng::LatLng;
+use s2::point::Point;
+use s2::rect::Rect;
+use s2::region::RegionCoverer;
+use s2::s1::angle::{Angle, Rad};
+use s2::s1::chordangle::ChordAngle;
+use std::vec::Vec;
+
+const LEVEL_MOD: u8 = 1;
+const EARTH_RADIUS_M: f64 = 6_371_010.0;
+
+fn earth_tangent_m_to_rad(meters: f64) -> f64 {
+    2.0 * (0.5 * meters / EARTH_RADIUS_M).atan()
+}
+
+fn lat_lng_from_ecef(ecef: ECEF<f64>) -> LatLng {
+    let wgs = WGS84::from(ecef);
+    let lat = Angle::from(Rad(wgs.latitude()));
```

Comment:
Nit: Do you think it would be helpful to have WGS LatLng to S2 LatLng conversion be a function?

Reply:
An alternative would be to have a `ToS2LatLng` trait and call its conversion function here

Reply:
I like the idea of a trait, implemented it...